src/d3d11/d3d11_swapchain.cpp

```diff
@@ -410,9 +451,11 @@ namespace dxvk {
           uint32_t                Repeat) {
     auto lock = pContext->LockContext();
 
-    // Bump frame ID as necessary
-    if (!Repeat)
-      m_frameId += 1;
+    if (!Repeat) {
+      m_frameId = (m_presenter->lowLatencyEnabled() && m_device->getLatencyMarkers().present) ?
```

Comment:
Two things mandates that m_frameId is monotonically increasing:
- **VK_KHR_present_id**: A non-zero presentId must be greater than any non-zero presentId passed previously by the application for the same swapchain.
- **CallbackFence**: The helper works similarly to a timeline semaphore so the timeline value must be increasing.

I think we need to at least ensure that the value doesn't decrease here. For present IDs, recreating the swapchain will do the job. CallbackFence will need some more modification: I'll describe my idea below.

- Make D3D9Swapchain use CallbackFence and remove Fence; the latter seems to be a superset of the former.
- Move CallbackFence into Presenter, so that it can be easily recreated by the Presenter. Change accesses to e.g. `D3D11Swapchain::m_frameLatencySignal` to something like `m_presenter->frameLatencySignal()`.
- Make `Presenter::destroySwapchain` destroy the signal too, then also recreate the signal in `Presenter::recreateSwapchain`. (`destroySwapchain` waits for the current frame to be signaled.) Add a new parameter to `Presenter::recreateSwapchain` for the frame ID the signal should be initialized with. Also reset `m_lastFrameId` to the specified value.

cc @doitsujin in case they have feedback for the proposed refactor.